pep-3156.txt

```diff
@@ -147,6 +148,15 @@ wait for a Future to complete by adding a callback to the Future.
 Likewise, the scheduler offers an operation to suspend a coroutine
 until a callback is called.
 
+In case of any other async framework cannot just use Future and Task
+as is for some reason the framework may reimplement
+``loop.create_future()`` and ``loop.create_task()`` calls by returning
+an object which implements a superset of Future/Task interfaces but
+adds a new functionality required by the framework.
+
+``loop.set_task_factory()`` may be useful as well for adopting asyncio
```

Comment:
From my understanding `set_task_factory()` was invited for overriding the current task implementation without total loop rewriting.
@1st1 IIRC you are the author of this change. I believe you can shed a light on design thoughts.